src/proper.erl

```diff
@@ -1173,6 +1272,45 @@ inner_test(RawTest, Opts) ->
 	false -> ShortResult
     end.
 
+%% @private
+%% @doc Runs PropEr in parallel mode, through the use of workers to perform the tests.
+%% Under this mode, PropEr will detect if a property is pure or impure based on the passed options.
+%% On the later case, it will start a node for every worker that will be spawned in order to
+%% avoid test collisions between them.
+-spec parallel_perform(test(), opts()) -> imm_result().
+parallel_perform(Test, #opts{property_type = pure, numtests = NumTests,
+                             numworkers = NumWorkers, strategy_fun = StrategyFun} = Opts) ->
+    TestsPerWorker = StrategyFun(NumTests, NumWorkers),
```

Comment:
I would suggest to factor out all lines between
```erlang
    TestsPerWorker = StrategyFun(NumTests, NumWorkers),
    ....
    ok = maybe_stop_cover_server([]),
```
into a separate function and use it both for this clause and the next one (by passing it `[]` or `NodeList`).

This will help both for maintainability and for understanding what is common and what is not.